src/__tests__/join.spec.ts

```diff
@@ -0,0 +1,11 @@
+import { join } from '../';
+
+describe('join', () => {
+  it('does some basic resolving', () => {
+    expect(join('/foo/bar', '..', 'baz')).toEqual('/foo/baz');
+  });
+
+  it('treats URLs as file URIs', () => {
+    expect(join('https://foo.test', 'com', 'baz')).toEqual('/com/baz');
```

Comment:
```suggestion
    expect(join('https://foo.test', 'com', 'baz')).toEqual('https://foo.test/com/baz');
```

Reply:
> treats URLs as file URIs

Yeah in general I think the premise of the test is wrong. URLs should be treated as URLs. File paths as file paths.

Should be able to pass the following in (prob not comprehensive, but ya get the idea):

- `join('https://foo.test', 'com', 'baz')`
- `join('https://foo.com/pets', '..', 'users', 123)`
- `join('/foo/bar', '..', 'baz')`
- `join('c:/foo/bar', '..', 'baz')`
- `join('c:/foo\bar', '..', 'baz')`